src/ensemble/basic_ensemble_solve.jl

```diff
@@ -23,6 +23,10 @@ $(TYPEDEF)
 """
 struct EnsembleSerial <: BasicEnsembleAlgorithm end
 
+function merge_stats(us)
+    mapreduce(x -> x.stats, merge, us)
```

Comment:
This is not safe for if stats is nothing

Reply:
Also, this merge is not defined for most stats types?

Reply:
I think what we should do is just move the diffeq stats object to SciMLBase and then add the merge function here. That would put it more inline with how the other bits are developing, like https://github.com/SciML/SciMLBase.jl/blob/master/src/solutions/nonlinear_solutions.jl#L2-L18. Since it doesn't require downstream functionality and is instead part of the general interface, it's a bit weird that Stats (which should be DEStats) is not in SciMLBase but is instead in DiffEqBase. That's just an organizational mistake. Can you make that swap as part of this PR?

Reply:
That seems quite a delicate dance of breaking changes and version bumps/pins all over SciML that I'm not sure I'm the right person to make. 

https://github.com/search?q=org%3ASciML+%22DiffEqBase.Stats%22&type=code

Handling the nothing case and method error is done though.